script/update-archive-version.ts

```diff
@@ -34,17 +53,26 @@ async function updateArchiveVersion() {
     osWebLoader
   );
 
-  const newBookLoader = makeUnifiedBookLoader(
-    createArchiveLoader(args.newArchive, {
-      archivePrefix: ARCHIVE_URL,
-    }),
-    osWebLoader
-  );
+  const newBookLoader = updatePipeline
+    ? makeUnifiedBookLoader(
+        createArchiveLoader(newArchiveUrl, {
+          archivePrefix: ARCHIVE_URL,
+        }),
+        osWebLoader
+      )
+    : currentBookLoader;
 
   const updateRedirectsPromises: Array<() => Promise<[BookWithOSWebData, number]>> = [];
-  const bookEntries = Object.entries(BOOKS);
 
   console.log('Preparing books...');
+  for (const book of booksToUpdate) {
+    await processBookVersionUpdate(book, updatePipeline ? newArchiveUrl : undefined);
+  }
+  const bookEntries = updatePipeline
+    ? Object.entries(BOOKS_CONFIG)
+    : Object.entries(BOOKS_CONFIG).filter((entry) =>
+      booksToUpdate.map((b) => b.bookId).find((id) => id === entry[0])
+    );
```

Comment:
@TomWoodward maybe you can suggest a cleaner way to filter the config for books on the booksToUpdate list